govcd/api_vcd_test.go

```diff
@@ -131,7 +134,30 @@ func (vcd *TestVCD) SetUpSuite(test *C) {
 		panic(err)
 	}
 	// creates a new VApp for vapp tests
-	vcd.vapp = *NewVApp(&vcd.client.Client)
+	vapp, err := vcd.createTestVapp("go-vapp-tests")
```

Comment:
Is this creation necessary? Can it be deferred to when a test requires such a vApp? 
If users are required to pass an existing vApp, then we should create a new one only when running destructive tests.